osbs/build/build_requestv2.py

```diff
@@ -0,0 +1,292 @@
+        self.explicit_build_node_selector = kwargs.get('explicit_build_node_selector', {})
+        self.auto_build_node_selector = kwargs.get('auto_build_node_selector', {})
+        self.isolated_build_node_selector = kwargs.get('isolated_build_node_selector', {})
+
+        logger.debug("setting params '%s' for %s", kwargs, self.user_params)
+        self.user_params.set_params(**kwargs)
+
+    # Override
+    @property
+    def inner_template(self):
+        raise RuntimeError('inner_template not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def customize_conf(self):
+        raise RuntimeError('customize_conf not supported in BuildRequestV2')
+
+    # Override
+    @property
+    def dj(self):
+        raise RuntimeError('DockJson not supported in BuildRequestV2')
+
+    def set_reactor_config_map(self, api):
+        if not self.user_params.reactor_config_map.value:
+            return
+        custom = self.template['spec']['strategy']['customStrategy']
+
+        reactor_config_map = {
+            'name': 'REACTOR_CONFIG',
+            'valueFrom': {
+                'configMapKeyRef': {
+                    'name': self.user_params.reactor_config_map.value,
+                    'key': 'config.yaml'
+                }
+            }
+        }
+        custom['env'].append(reactor_config_map)
+        if api:
+            cm_response = api.get_config_map(self.user_params.reactor_config_map.value)
+            for key in _CM_SECRETS:
+                self.template['spec'][key] = cm_response.get_data_by_key(key)
+        else:
+            raise OsbsValidationException("reactor config map without an OSBS object")
+
+    def adjust_for_scratch(self):
+        """
+        Remove certain plugins in order to handle the "scratch build"
+        scenario. Scratch builds must not affect subsequent builds,
+        and should not be imported into Koji.
+        """
+        if self.scratch:
+            self.template['spec'].pop('triggers', None)
+            self.set_label('scratch', 'true')
+
+    def adjust_for_isolated(self):
+        if not self.isolated:
+            return
+
+        self.template['spec'].pop('triggers', None)
+
+        if not self.user_params.release.value:
+            raise OsbsValidationException('The release parameter is required for isolated builds.')
+
+        if not ISOLATED_RELEASE_FORMAT.match(self.user_params.release.value):
+            raise OsbsValidationException(
+                'For isolated builds, the release value must be in the format: {0}'
+                .format(ISOLATED_RELEASE_FORMAT.pattern))
+
+        self.set_label('isolated', 'true')
+        self.set_label('isolated-release', self.user_params.release.value)
+
+    def adjust_for_repo_info(self):
+        if not self._repo_info:
+            logger.warning('repo info not set')
+            return
+
+        if not self._repo_info.configuration.is_autorebuild_enabled():
```

Comment:
`if not x: ... else: ...` is extra mental work to parse. Can we switch it around so it's `if x: ... else: ...`?

Reply:
Fixed